File: Day-31/344A- Magnets.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    int n;
    cin >> n;
    string prev;
    cin >> prev;
    int groups = 1;
    for (int i = 1; i < n; i++) {
        string current;
        cin >> current;
        if (prev[1] == current[0]) {
            groups++;
        }
        prev = current;
    }
    cout << groups << endl;
    return 0;
}

File: Day-31/344A- Magnets.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        String prev = scanner.next();
        int groups = 1;
        for (int i = 1; i < n; i++) {
            String current = scanner.next();
            if (prev.charAt(1) == current.charAt(0)) {
                groups++;
            }
            prev = current;
        }
        System.out.println(groups);
    }
}
